hlc/plan/planner.py

```diff
@@ -46,48 +54,43 @@ def layer_pos(grid_dimension, layer):
     return left_layer, right_layer, top_layer, bottom_layer
 
 
-def update_pos(pos, action):
-    new_pos = add_pos_tuple(pos, action)
-    return new_pos
-
-
-def plan(grid, grid_dimension, start_pos = (0,0), layer = 0, orientation = FORWARD):
-    pos = start_pos
+def plan(grid_dimension, start_pos = (0, 0), layer = 0, orientation = FORWARD):
+    pos = Position(start_pos)
     plan =  []
 
     max_layer = max(grid_dimension) // 2
 
     while layer < max_layer:
 
         plan.append(MOVE_FORWARD)
-        pos = update_pos(pos, orientation)
+        pos.update(orientation)
 
-        if pos == add_pos_tuple(start_pos, (1, 0)):
+        if pos.current() == Position(start_pos).update((1, 0)):
```

Comment:
Try overwriting the `__eq__`-function in the Position class so you don't have to convert the startposition to a tuple each time you do a compare.

Reply:
Also related to line 73

Reply:
I do not convert the `start_pos` to a tuple. It already is. But since I need to take the point to the right, `(1, 0)`, I need to update the `start_pos`.